lib/normalize_grammar.re

```diff
@@ -1,5 +1,78 @@
 open Common;
 
-let normalize = _ast =>
-  raise (Todo);
-  
\ No newline at end of file
+let normalize_to_simple = body => {
+  switch (body) {
+  | Ast_grammar.TOKEN =>
+    Ast_grammar_normalized.ATOM(
+      Ast_grammar_normalized.TOKEN
+    );
+  | Ast_grammar.SYMBOL(name)=>
+    Ast_grammar_normalized.ATOM(
+      Ast_grammar_normalized.SYMBOL(name)
+    );
+  /* Ignore rest */
+  | _ => {
+    print_string("Inside normalize_to_simple\n");
+    raise(Todo);
+  }
+  }
+}
+
+let normalize_to_atom = body => {
+  switch (body) {
+  | Ast_grammar.TOKEN =>
+    Ast_grammar_normalized.TOKEN
+  | Ast_grammar.SYMBOL(name)=>
+    Ast_grammar_normalized.SYMBOL(name)
+  /* Ignore rest */
+  | _ => {
+    print_string("Inside normalize_to_atom\n");
+    raise(Todo);
+  }
+  }
+}
+
+let normalize_body = (rule_body) => {
+  switch(rule_body) {
+  | Ast_grammar.TOKEN =>
+    Ast_grammar_normalized.SIMPLE(
+      Ast_grammar_normalized.ATOM(
+        Ast_grammar_normalized.TOKEN
+      )
+    );
+  | Ast_grammar.SYMBOL(name) =>
+    Ast_grammar_normalized.SIMPLE(
+      Ast_grammar_normalized.ATOM(
+        Ast_grammar_normalized.SYMBOL(name)
+      )
+    );
+  | Ast_grammar.SEQ(bodies) =>
```

Comment:
here you can get intermediates, so you'll need
let xs = List.map(normalize_to_atom, bodies);
let atoms = List.map(fst, xs);
let intermediates = List.flatten(List.map(snd, xs));
(B.SEQ(atoms), intermediates)

Reply:
Ok.